cmd/ears/run.go

```diff
@@ -0,0 +1,85 @@
+package main
+
+import (
+	"fmt"
+	"github.com/rs/zerolog"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"github.com/xmidt-org/ears/internal/pkg/app"
+	"github.com/xmidt-org/ears/internal/pkg/cli"
+	"go.uber.org/fx"
+)
+
+var runCmd = &cobra.Command{
+	Use:   "run",
+	Short: "Runs the EARS microservice",
+	Long:  `Runs the EARS microservice`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		logger, err := InitGlobalLogger()
+		if err != nil {
+			log.Logger.Error().Str("op", "InitLogger").Msg(err.Error())
+			os.Exit(1)
+		}
+		earsApp := fx.New(
```

Comment:
I can foresee an issue here.  The API server will need a reference to the application.  More specifically, the handlers attached to the mux will need access to the application.  HTTP handlers are essentially functions that transform API input models to some application model, and then translate the output from the application to some API model.  They still need to make a call against the application.

We should also take a hard look at go-kit.  You can see how they break down a service into functions that process input and output and then call the service (which is a well defined interface, not an actual struct)  https://github.com/go-kit/kit/blob/master/examples/stringsvc1/main.go

Reply:
So the uber/fx app should just handle dependency injections and lifecycles. I don't think we wan to reference it. We should have a different interface (for the sake of argument, lets call it earsService) that we can pass around. API server can have a reference to that through a uberfx provider (ie NewEarsService)

Reply:
Maybe this example shows more clearly what I'm talking about.  We have an API for adding a route.  We have a handler to add the route. The handler must call some sort of thing with a method `AddRoute(...)` on it.  How does it access that thing?

```
func addRoute(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()

	log.Ctx(ctx).Debug().Msg("addingRoute")

       // Somehow add the route
       // Somehow have to reference the `someThing` with the `AddRoute` method on it
       routeInfo := someThing.AddRoute( // arguments extracted from the request object )

	resp := Response{
		Item: {
			RouteId: routeInfo.ID,
		,
	}
	resp.Respond(w)
}
```

Reply:
I think there are many way to structure this. It is up to us. One idea would be to have a yet to be implemented RoutingService (with a uberfx provider called NewRoutingService) as the someThing that you mentioned above. We can use uberfx to inject the routingService instance to the API servers. 

More concretely, we can inject the routingService here: https://github.com/xmidt-org/ears/blob/bare-bone/internal/pkg/app/handlers_v1.go#L9

```
var routingService RoutingService
func NewRouter(srv RoutingService) (*mux.Router, error) {
  routingService = srv
}
```

Reply:
@holidaymike  It seems like the code you referenced would be the place to do this.  I think since this is the bare bones branch, it probably makes sense to include it in this PR with a mock example.

I'm still curious when the routes are attached, how it gets access to `routingService`. For instance, is the handler a closure that then calls `routingService`?  Or if it's a separate function call (like `addRoute` in the above example), how does the `routingService` object get to the handler so that it can be referenced.

Reply:
@gtrevg Please review. I copy @boriwo RoutingTableManager interface over partially just to demonstrate how I think we can hook this up.